pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
> But won't that make totals over say a day very inaccurate (especially with server restarts, etc)? or if the rate varies a lot.

1 login per second on average per day is exactly the same as 86400 logins per day.

> Which sounds like a very bad thing if you want to sum events. But maybe it becomes more accurate than I think due to the way things are stored in Prometheus?

It's just a different way of saying the same thing.

> And having roll ups to do long term trending is what I really like about TSDBs as well so that's why I have always liked deltas more.

In terms of getting the initial data into the TSDB, raw counters are better. Handling the data after that is a different question.

> But looks like Prometheus makes working with raw counters a bit easier than say Graphite or InfluxDB

`nonNegativeDerivative` isn't as good as Prometheus at handling resets, but it's not terrible.